src/raw/mod.rs

```diff
@@ -787,7 +791,50 @@ impl<'f, 'a, A: Automaton> Streamer<'a> for Stream<'f, A> {
     type Item = (&'a [u8], Output);
 
     fn next(&'a mut self) -> Option<Self::Item> {
-        self.0.next().map(|(inp, out, _)| (inp, out))
+        if let Some(out) = self.0.empty_output.take() {
+            if self.0.end_at.exceeded_by(&[]) {
+                self.0.stack.clear();
+                return None;
+            }
+            let start = self.0.aut.start();
+            if self.0.aut.is_match(&start) {
+                return Some((&[], out));
+            }
+        }
+        while let Some(state) = self.0.stack.pop() {
+            if state.trans >= state.node.len()
+                    || !self.0.aut.can_match(&state.aut_state) {
+                if state.node.addr() != self.0.fst.root_addr {
+                    self.0.inp.pop().unwrap();
+                }
+                continue;
+            }
+            let trans = state.node.transition(state.trans);
+            let out = state.out.cat(trans.out);
+            let next_state = self.0.aut.accept(&state.aut_state, trans.inp);
+            let is_match = self.0.aut.is_match(&next_state);
+            let next_node = self.0.fst.node(trans.addr);
+            self.0.inp.push(trans.inp);
+            self.0.stack.push(StreamState {
+                trans: state.trans + 1, .. state
+            });
+            self.0.stack.push(StreamState {
+                node: next_node,
+                trans: 0,
+                out: out,
+                aut_state: next_state,
+            });
+            if self.0.end_at.exceeded_by(&self.0.inp) {
+                // We are done, forever.
+                self.0.stack.clear();
+                return None;
+            }
+            if next_node.is_final() && is_match {
+                let out = out.cat(next_node.final_output());
+                return Some((&self.0.inp, out));
+            }
+        }
+        None
```

Comment:
I don't like code duplication, the only thing different with the `StreamWithState` is the Clone of the `aut_state`.